apps/web/src/components/templates/in-app-editor/InAppEditorBlock.tsx

```diff
@@ -70,3 +94,20 @@ function ContentContainerController({
     </>
   );
 }
+
+function AvatarContainerController({ control, index }: { control: Control<IForm>; index: number }) {
+  return (
+    <>
+      <Controller
+        name={`steps.${index}.template.avatarDetails` as any}
+        data-test-id="in-app-avatar-item"
+        control={control}
+        render={({ field }) => {
+          const { ref, ...fieldRefs } = field;
+
+          return <AvatarContainer {...fieldRefs} />;
+        }}
+      />
+    </>
```

Comment:
I feel like the core here will be a little better if you will use the `useController` hook and just destructure the `value, onChange` props that you will pass to `AvatarContainer`.
Also, no need for `Fragment`, and the `data-test-id` is not used.